cookbooks/fb_consul/recipes/default.rb

```diff
@@ -0,0 +1,157 @@
+#
+# Cookbook:: fb_consul
+# Recipe:: default
+#
+# Copyright (c) 2020-present, Vicarious, Inc.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+node.default['fb_iptables']['filter']['INPUT']['rules']['consul']['rules'] = [
+  # Gossip protocol between agents and servers
+  '-p tcp --dport 8301 -j ACCEPT',
+  '-p udp --dport 8301 -j ACCEPT',
+]
+
+node.default['fb_users']['users']['consul'] = {
+  'comment' => 'consul agent user',
+  'home' => '/run/consul',
+  'shell' => '/usr/sbin/nologin',
+}
+
+package 'consul' do
+  action :upgrade
+  notifies :restart, 'service[consul]'
+end
+
+whyrun_safe_ruby_block 'validate config' do
+  block do
+    node['fb_consul']['config'].each_key do |key|
+      if ['config-file', 'config-dir'].include?(key)
+        fail "fb_consul::default: #{key} is not allowed in 'config'! " +
+          'Please use "services" and "checks" to configure consul.'
+      end
+    end
+  end
+end
+
+directory 'consul data dir' do
+  only_if do
+    node['fb_consul']['enabled']
+  end
+  path lazy { node['fb_consul']['config']['data_dir'] }
+  owner 'consul'
+  group 'root'
+  mode '0770'
+end
+
+execute 'fix_perms' do
+  command 'chown -R consul /var/lib/consul'
+  action :nothing
+end
+
+file '/etc/systemd/system/consul.service' do
```

Comment:
Why?

Reply:
because the package doesn't come with one.

Reply:
So where does this one come from?

Reply:
oh I see, because it's not a cookbook file. Huh. Let me go poke.

Reply:
Oh. This is left-over from when we were using the very old package that didn't have one, which is why we're deleting that. Will nuke